src/Kestrel.Core/Internal/Http/HttpProtocol.cs

```diff
@@ -899,7 +899,7 @@ private Task WriteChunkedAsync(ArraySegment<byte> data, CancellationToken cancel
 
         private static void WriteChunk(PipeWriter writableBuffer, ArraySegment<byte> buffer)
         {
-            var writer = OutputWriter.Create(writableBuffer);
+            var writer = new BufferWriter(writableBuffer);
```

Comment:
`BufferWriter` will throw an exception if `buffer.Count` > max `MemoryPool` block size?

When its Span isn't big enough to write to it asks for a larger span via  [`Resize`](https://github.com/dotnet/corefxlab/blob/38f273b5383c2eb6a72d7137ce527af484e928d2/src/System.Buffers.Experimental/System/Buffers/Text/BufferWriter.cs#L49-L59) and then copies over the previously written data.
```csharp
private void Resize(int desiredBufferSize = 0)
{
    if (Enlarge == null) throw new BufferTooSmallException();
    // double the size; add one to ensure that an empty buffer still resizes
    if (desiredBufferSize <= 0) desiredBufferSize = _buffer.Length * 2 + 1;
    else if (desiredBufferSize < _buffer.Length + 1) throw new ArgumentOutOfRangeException(nameof(desiredBufferSize));
    var newBuffer = Enlarge(desiredBufferSize).Span;
    if (newBuffer.Length <= _buffer.Length) throw new Exception("Enlarge delegate created too small buffer");
    Written.CopyTo(newBuffer);
    _buffer = newBuffer;
}
```
If its already the max size for the pool it will throw an exception?

/cc @KrzysztofCwalina @aahsonkhan

Reply:
`OutputWriter` would probably be better here as it `Advance`s the pipe leaving previously written data when it runs out of `Span`, then gets a new `Span` for the rest (and repeats)

Reply:
Nevermind, is a different `BufferWriter`